### ISSUE TITLE
This in module not undefined

### ISSUE THREAD

**Issue by raulsebastianmihaila** [webpack/webpack#5074]
**Do you want to request a *feature* or report a *bug*?**
Report a bug.

**What is the current behavior?**
The value of `this` at the top level in a module is not `undefined`; instead it's an object.
**If the current behavior is a bug, please provide the steps to reproduce.**
```js
export default 4;

console.log('module this', this);
```

**What is the expected behavior?**
`this` at the top level in the module must be `undefined`: https://tc39.github.io/ecma262/#sec-module-environment-records-getthisbinding
**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
webpack 2.6.1, node 6.11.0, ubuntu

**Comment by raulsebastianmihaila** [comment]
Note I'm not using any loaders.

**Comment by ooflorent** [comment]
This is the expected behavior. ES modules are strict. So `this` is `undefined`.
See [Value of `this` at top level](http://exploringjs.com/es6/ch_modules.html#_browsers-scripts-versus-modules).

**Comment by ooflorent** [comment]
You're right. I can reproduce the issue. The runtime creates each module using:

```js
modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
                       ^^^^^^^^^^^^^^
```

A possible fix is to add a flag to each module and change the context of the factory:

```js
modules[moduleId].call(isHarmonyModule ? undefined : module.exports, module, module.exports, __webpack_require__);
```

**Comment by ooflorent** [comment]
Doing this is quite challenging with the current runtime. Another solution is the replace `this` by `undefined`.

**Comment by ooflorent** [comment]
Fixed in `4.0.0-alpha.5`.
